pkg/provider/provider.go

```diff
@@ -567,3 +608,12 @@ func parseCheckpointObject(obj resource.PropertyMap) (oldInputs, live *unstructu
 	live = &unstructured.Unstructured{Object: liveMap.(map[string]interface{})}
 	return
 }
+
+func initializationError(id string, err error, inputsAndComputed *structpb.Struct) error {
+	detail := pulumirpc.ErrorResourceInitFailed{
+		Id:         id,
+		Properties: inputsAndComputed,
+		Reasons:    []string{err.Error()},
+	}
+	return rpcerror.WithDetails(rpcerror.New(codes.Unknown, err.Error()), &detail)
```

Comment:
I think we should probably avoid using `Unknown` in general, but we use it so pervasively it's probably not worth the time and effort to clean up.

Reply:
So the reason we're returning `codes.Unknown` is because returning a code that makes "more sense" would cause us to return `status.OK`, per [this code](https://github.com/pulumi/pulumi/blob/ad3b5e7ee88346bc7e960de9f953957a72f84516/pkg/resource/plugin/provider_plugin.go#L574).

This is obviously gross, but I was not comfortable changing this code, because it seems like our provider depends on this behavior, and changing our protocol did not seem like a good thing to do when introducing partial update.

I've filed https://github.com/pulumi/pulumi/issues/1615 to follow up.